lib/services/platform-service.ts

```diff
@@ -368,15 +368,56 @@ export class PlatformService implements IPlatformService {
 				this.$errors.failWithoutHelp("Unable to find built application. Try 'tns build %s'.", platform);
 			}
 
-			this.$fs.ensureDirectoryExists(path.dirname(targetPath)).wait();
+			function ext(file: string): string {
+				return path.extname(file).toLowerCase();
+			}
+			function targetExt(file: string): string {
+				let e = ext(file);
+				return (e === ".ipa" || e === ".app") ? e : ".";
+			}
+
+			switch(ext(packageFile) + "->" + targetExt(targetPath)) {
+				case ".ipa->.ipa":
+					this.$logger.trace(`Copying file '${packageFile}' to '${targetPath}'...`);
+					this.$fs.copyFile(packageFile, targetPath).wait();
+					this.$logger.info(`Copied file '${packageFile}' to '${targetPath}'.`);
+					break;
+
+				case ".app->.app":
+					this.$logger.trace(`Copying directory '${packageFile}' to '${targetPath}'...`);
+					shell.cp("-Rf", packageFile + "/*", targetPath);
+					this.$logger.info(`Copied directory '${packageFile}' to '${targetPath}'.`);
+					break;
+
+				case ".ipa->.":
+				case ".app->.":
+					this.$fs.ensureDirectoryExists(path.dirname(targetPath)).wait();
+					if (!this.$fs.exists(targetPath).wait() || !this.$fs.getFsStats(targetPath).wait().isDirectory()) {
+						this.$logger.trace(`Expected directory for --copy-to at ${targetPath}`);
```

Comment:
in case I write:
```
$ tns build ios --emulator --copy-to /Users/vladimirov/test/myFile.txt`
```
this message will not be shown, but I'll also not receive my moved file. Consider using 
```TypeScript
this.$logger.warn("...");
```
or even
```TypeScript
this.$errors.failWithoutHelp("...");
```